packages/asc-ui/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react'
+import { ChevronLeft, ChevronRight } from '@datapunt/asc-assets'
+import PaginationStyle, {
+  Props,
+  PageNumber,
+  PreviousButton,
+  NextButton,
+} from './PaginationStyle'
+
+const Pagination: React.FC<Props & React.HTMLAttributes<HTMLElement>> = ({
+  page = 1,
+  collectionSize,
+  pageSize = 10,
+  onPageChange,
+  ...otherProps
+}) => {
+  const [currentPage, setCurrentPage] = React.useState<number>(page)
+  const totalPages = Math.ceil(collectionSize / pageSize)
+  const pageCluster = [currentPage]
+  if (currentPage > 2) {
+    pageCluster.unshift(currentPage - 1)
+  }
+  if (totalPages <= 5 || currentPage < totalPages - 1) {
+    pageCluster.push(currentPage + 1)
+  }
+
+  React.useEffect(() => {
+    if (onPageChange !== undefined) onPageChange(currentPage)
+  }, [currentPage, onPageChange])
+
+  const handlePrevious = () => {
+    setCurrentPage(currentPage - 1)
+  }
+
+  const handleNext = () => {
+    setCurrentPage(currentPage + 1)
+  }
+
+  const renderButton = (pageNumber: number) => {
```

Comment:
We should never add a component inside of a component, as this would cause major performance issues. Best is to introduce a separate component (could be inside this file tho) called something like `PaginationButton` where it accepts a `pageNumber`

Reply:
In this case I'm not sure if the 60 lines of additional code are a lesser evil than the (in this instance) small performance overhead. But I've refactored as you suggested